bears/general/KeywordBear.py

```diff
@@ -2,7 +2,63 @@
 
 from coalib.bearlib import deprecate_settings
 from coalib.bears.LocalBear import LocalBear
+from coalib.results.Diff import Diff
 from coalib.results.Result import RESULT_SEVERITY, Result
+from coalib.results.SourceRange import SourceRange
+
+from bears.general.AnnotationBear import AnnotationBear
+
+
+def _get_comments(dependency_results):
+    if not dependency_results:
+        return
+
+    annotation_bear_results = dependency_results.get('AnnotationBear')
+    if (not annotation_bear_results or
+            not isinstance(annotation_bear_results, list)):
+        return
+
+    for result in annotation_bear_results:
+        yield from result.contents.get('comments', [])
+
+
+def generate_diff(comments, file, filename,
+                  line, line_number, pos):
+    diffs = {}
+    todo_source_range = SourceRange.from_values(filename, line_number,
+                                                pos + 1)
+    for comment_sourcerange in comments:
+        if todo_source_range not in comment_sourcerange:
+            continue
+
+        comment_start = comment_sourcerange.start.column
+        comment_end = comment_sourcerange.end.column
+        in_multi_line_comment = (
+            comment_sourcerange.start.line != comment_sourcerange.end.line
+        )
+
+        line_before_todo_comment = line[:comment_start - 1].rstrip()
+        line_behind_todo_comment = line[comment_end:].rstrip()
+
+        line_replacement = line_before_todo_comment + line_behind_todo_comment
+
+        diff = Diff(file)
+        if line_replacement and not in_multi_line_comment:
+            diff.change_line(
+                line_number,
+                line,
+                line_replacement + '\n')
+        elif line_replacement and in_multi_line_comment:
+            text_replacement = line[pos:]
+            diff.change_line(
+                line_number,
+                line,
+                line.replace(text_replacement, '').rstrip() + '\n')
+        else:
+            diff.delete_line(line_number)
+
+        diffs[filename] = diff
```

Comment:
actually you could do a `return {filename: diff}` right here and a `return {}` in line 61 which would give a small performance benefit and simplifies the code (we can remove line 27). There should never be more than one source range touching the keyword, right? And if there is more than one (I could think of some nasty corner cases) this loop would yield conflicting patches.

We can also remove the for loop in favour of a filter: do a filter, assert the length of 1 or 0 (exception otherwise), use the first element if it's there, return `{}` otherwise.

This should be a very simple&quick change